app/src/main/java/de/westnordost/streetcomplete/quests/toilet_availability/AddToiletAvailability.java

```diff
@@ -20,9 +20,9 @@
 
 	@Override protected String getTagFilters()
 	{
-		// only for malls and big stores because users should not need to go inside a non-public
+		// only for malls, big stores and rest areas because users should not need to go inside a non-public
 		// place to solve the quest. (Considering malls and department stores public enough)
-		return "nodes, ways with shop ~ mall|department_store and name and !toilets";
+		return "nodes, ways with (shop ~ mall|department_store and name and !toilets) or (highway ~ services|rest_area and !toilets)";
```

Comment:
BTW the query can be optimized if you put the "and not toilets" outside of everything else. It applies to both things, of course.